### PR TITLE
test: add tests for duplicate prevention (fixes #2952)

### LINKED ISSUE #2952: Add tests for the duplicate entries in the controller exported interfaces
Add missing tests for the fix introduced in the PR: https://github.com/ros-controls/ros2_control/pull/2951, where the resource manager gets duplicate entries while activating controller due to multiple calls from `make_controller_exported_state_interfaces_available`

### PR DESCRIPTION
**Pull request by kamal2730**
Add tests for PR #2951 - duplicate entries fix in controller exported interfaces. This adds tests for both reference and state interfaces to ensure that calling make_controller_reference_interfaces_available() and make_controller_exported_state_interfaces_available() multiple times does not create duplicate entries.
- Extended managing_controllers_reference_interfaces with duplicate prevention test
- Added new managing_controllers_state_interfaces test with duplicate prevention
- Both tests verify no duplicates are created when making interfaces available multiple times (simulating multiple activate calls) Fixes: ros-controls/ros2_control#2952
See-also: ros-controls/ros2_control#2951

To send us a pull request, please:

- [x] Fork the repository.
- [x] Modify the source; please focus on the specific change you are contributing. If you also reformat all the code, it will be hard for us to focus on your change.
- [x] Ensure local tests pass. (`colcon test` and `pre-commit run` (requires you to install pre-commit by `pip3 install pre-commit`)
- [x] Commit to your fork using clear commit messages.
- [x] Send a pull request, answering any default questions in the pull request interface.
- [x] Pay attention to any automated CI failures reported in the pull request, and stay involved in the conversation.

### CODE REVIEW

File: hardware_interface_testing/test/test_resource_manager.cpp [hunk 1]
```diff
@@ -1334,6 +1335,27 @@ TEST_F(ResourceManagerTest, managing_controllers_reference_interfaces)
   EXPECT_EQ(reference_interface_values[1], 2.0);
   EXPECT_EQ(reference_interface_values[2], 33.3);
 
+  // DUPLICATE PREVENTION TEST
+
+  // Make reference interfaces available again (simulating multiple activate calls)
+  rm.make_controller_reference_interfaces_available(CONTROLLER_NAME);
+
+  // Count before
+  size_t count_before = rm.available_command_interfaces().size();
```
Comment (saikishor) [inline hardware_interface_testing/test/test_resource_manager.cpp:1348]: Maybe we should make sure that after the call the interfaces size has changed

File: hardware_interface_testing/test/test_resource_manager.cpp [hunk 2]
```diff
@@ -1349,6 +1371,80 @@ TEST_F(ResourceManagerTest, managing_controllers_reference_interfaces)
...
+  // Make available AGAIN - this should NOT create duplicates
+  rm.make_controller_exported_state_interfaces_available(CONTROLLER_NAME);
+
+  // Count after - should be same as before
+  size_t count_after = rm.available_state_interfaces().size();
+  EXPECT_EQ(count_before, count_after) << "Duplicate state interfaces detected!";
+
+  // Verify all entries are unique
+  EXPECT_TRUE(ros2_control::is_unique(rm.available_state_interfaces()));
+
```
Comment (saikishor) [inline hardware_interface_testing/test/test_resource_manager.cpp:1456]:
```suggestion
  for (const auto & interface : FULL_STATE_INTERFACE_NAMES)
  {
    EXPECT_TRUE(rm.state_interface_exists(interface));
    EXPECT_TRUE(rm.state_interface_is_available(interface));
  }
```


It's nice to also check it again, because this way we can also check that calling available again, won't make it unavilable without affecting the size

**Review by saikishor — CHANGES_REQUESTED** [review]
Just some minor nits

**Review by saikishor — APPROVED** [review]

**Comment by kamal2730** [comment]
Hello @christophfroehlich , this PR is ready for review. Kindly review it when you have time.

**Review by christophfroehlich — APPROVED** [review]
